support/ci/promote_rc.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+set -e
+
+if [ -z ${HAB_AUTH_TOKEN+x} ]; then
+  echo "HAB_AUTH_TOKEN var is unset"
+  exit 1
+fi
+if [ -z ${BINTRAY_USER+x} ]; then
+  echo "BINTRAY_USER var is unset"
+  exit 1
+fi
+if [ -z ${BINTRAY_KEY+x} ]; then
+  echo "BINTRAY_KEY var is unset"
+  exit 1
+fi
+
+HAB_VERSION=$(cat VERSION)
+PROMOTE_CHANNEL=stable
+RC_CHANNEL=rc-$HAB_VERSION
+RC_PKGS_JSON=$(curl https://bldr.habitat.sh/v1/depot/channels/core/$RC_CHANNEL/pkgs | jq -r '.data' | jq -c '.[]')
```

Comment:
Digging further, it looks like this will return all releases in the channel, not just the latest ones. In the ideal case, there will only be a single release for each package in this release channel, but if we have issues with the release and end up having to re-run part or all of it, we can end up with multiple releases for a given package.

Ideally there'd be a flag for this in the API, but barring that, we can probably leverage a bit of shell scripting to take care of things for us, likely with a bit of `awk` and `sort`. It may be easier to use `jq` to get the data into a form that's easier for those tools to interact with:

```sh
curl https://bldr.habitat.sh/v1/depot/channels/core/stable/pkgs | \
jq -cr '.data |
      map ([.origin +"/"+ .name +"/"+ .version +"/"+ .release, .origin, .platforms[0]]) |
      map(join(" ")) |
      .[]'
```
That would yield something like: 
```
core/7zip/16.04/20170209152224 core x86_64-windows
core/R/3.4.0/20170623211014 core x86_64-linux
core/R/3.4.2/20171011183236 core x86_64-linux
core/R/3.4.2/20171013091650 core x86_64-linux
core/R core/R/3.4.2/20171014220034 core x86_64-linux
...
```

Reply:
Though I doubt we'll be running afoul of the pagination limit anytime soon, it may be prudent to at least check the result that comes back to ensure that we are indeed staying on one page of results.